src/buildstream/utils.py

```diff
@@ -359,6 +359,18 @@ def sha256sum(filename: str) -> str:
     return h.hexdigest()
 
 
+try:
+    os.copy_file_range  # type: ignore[attr-defined] Requires Python 3.8 or newer
+except AttributeError:
+    _copy_file = lambda src, dest: shutil.copyfile(src, dest)
```

Comment:
Why not simply
```suggestion
    _copy_file = shutil.copyfile
```
 ?

Reply:
Mypy complained the two functions have different amount of parameters.

Reply:
I rewrote the entire piece of code.